### PR TITLE
Add future.cancel() and fix worker send issue

### PR DESCRIPTION
**Pull request by vinok88**
Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/Strand.java [hunk 1]
```diff
@@ -52,6 +52,7 @@
     public boolean blockedOnExtern;
     public ChannelDetails[] channelDetails;
     private Map<String, Object> globalProps;
+    public boolean doCancel;
```
Comment (Kishanthan) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/Strand.java]: just "cancel" ?

Comment (vinok88) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/Strand.java]: Done. Thanks.

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/FutureValue.java [hunk 2]
```diff
@@ -72,4 +75,18 @@ public void stamp(BType type, List<TypeValuePair> unresolvedValues) {
...
+     public boolean isDone() {
+         return isDone;
+     }
+
+     public boolean isCancelled() {
+         return isCancelled;
+     }
+
+     public boolean cancel() {
+         this.strand.cancel = true;
```
Comment (Kishanthan) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/FutureValue.java]: why we need the strand also to have the cancel flag aprt from FutureValue having it?

Comment (vinok88) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/FutureValue.java]: FutureValue is the reference to a strand that others can use it to cancel the strand. Within functions, strand is the one that carry this meta info. We need this info within functions. A strand cannot access it's own future. Hence this cancel flag is used to track that there is a cancel requested.

**Pull request merged**